src/Components/LoginComponent/Signup.js

```diff
@@ -0,0 +1,319 @@
+import React, { useState } from "react";
+import styled, { ThemeProvider } from "styled-components";
+import { withRouter } from "react-router-dom";
+import Apiresource from "../../Config";
+
+
+function Signup(props, visible) {  
+    const [ userInfo, setName ] = useState({email:"",password:"",name:""});
+
+
+    const emailDetector = (e) => {
+        setName({...userInfo, [e.target.name]:e.target.value});
+        }; 
+
+
+    const submit = (userInfo) => {
```

Comment:
```suggestion
    const submit = ({ email, password, nickname }) => {
```
함수 파라미터 자리에서 직접 구조분해할당 가능합니다

```suggestion
    const submit = () => {
      const {email, password, nickname} = userInfo
```
=> 그런데 보아하니 state를 직접 사용하는 내용이라 인자로 넘겨 받을 필요 없을 것 같네요

Reply:
수정 진행하였습니다.